src/Hazelcast.Net/Clustering/RetryStrategy.cs

```diff
@@ -93,14 +88,14 @@ public async ValueTask<bool> WaitAsync(CancellationToken cancellationToken)
 
             var elapsed = (int) (DateTime.UtcNow - _begin).TotalMilliseconds;
 
-            if (elapsed > _timeout)
+            if (elapsed > _timeoutMilliseconds)
```

Comment:
Does that check work for the new default value (-1)? In Java, we set this to Long.MAX_VALUE if it is equal to -1 before constructing something similar to this class